### PR TITLE
Circumvents issue with endless refreshing on #messages tab

### PR DESCRIPTION
**Pull request by philippbayer**
Have a look - does it work on your ends too?

Could fix https://github.com/openSNP/snpr/issues/425

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -9,7 +9,6 @@
 //= require ./jqplot.highlighter.min.js
 //= require ./jqplot.pieRenderer.min.js
 //= require jquery_nested_form
-//= require bootstrap.min
```
Comment (tsujigiri) [inline app/assets/javascripts/application.js]: 😮

File: app/views/layouts/_navbar.html.erb [hunk 2]
```diff
@@ -43,7 +43,7 @@
           <li><%= link_to("Settings", "/users/"+current_user.id.to_s+"/edit", class: "navbar-menu__options", :id => current_user) %></li>
           <li><%= link_to("Edit Your Phenotypes", "/users/"+current_user.id.to_s+"/edit#phenotypes", class: "navbar-menu__options")%></li>
           <li class="divider"></li>
-          <li><%= link_to("Messages ("+current_user.messages.where(:user_has_seen => false).length.to_s+")","/users/" + current_user.id.to_s + "#messages", :method => "get", class: "navbar-menu__options")%></li>
+          <li><%= link_to "Messages (#{current_user.messages.where(:user_has_seen => false).length})","/users/#{current_user.id}#messages", class: 'navbar-menu__options', :data => { :reload => true }%></li>
```
Comment (tsujigiri) [inline app/views/layouts/_navbar.html.erb]: `"/users/#{current_user.id}#messages"` can be written as `user_path(current_user, anchor: 'messages')`. And I suppose the `:data => { :reload => true }` might have something to do with the reloading? :smile:

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (philippbayer) [reply app/assets/javascripts/application.js]: interestingly, removing this causes no problem on my end - I do get an endless 404 when it's kept in

File: app/views/layouts/_navbar.html.erb (hunk 2 above)
Comment (philippbayer) [reply app/views/layouts/_navbar.html.erb]: Yeah the :method => "get" causes an endless reload on my end (right now on production too), Firefox 62.0.3 - when I force the reload then the entire page reloads, but only once! and it lands on the correct tab

Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: Which `:method => "get"`? I don't think it has anything to do with the link. It happens if you put `https://www.opensnp.org/users/15#messages` into your address bar as well. [This](https://github.com/openSNP/snpr/blob/a96b015fcfa1bed6aa23d33739beeceb67d55741/app/assets/javascripts/application.js#L51-L57) looks a bit suspicious.

Comment (philippbayer) [reply app/views/layouts/_navbar.html.erb]: By method => get I mean the current way it's implement on production - 
That JS snippet is definitely the main 'suspect', getting stuck in an endless loop - let me fix that. My forced reloading seems to not have triggered that JS, thereby circumventing it (on my system?)

Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: I think what's missing in the JS is comparing `hash` with `anchor` and only redirecting if they don't match.

Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: Ah, no. Scratch that. It needs to figure out, whether the tab is active.

Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: What's weird is, that it only does that on the messages tab in the first place. :thinking:

Comment (philippbayer) [reply app/views/layouts/_navbar.html.erb]: and why does this break only know? I put it in a year ago :P

Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: I have no idea. 😅

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (philippbayer) [reply app/assets/javascripts/application.js]: I've added that now back in

File: app/views/layouts/_navbar.html.erb (hunk 2 above)
Comment (tsujigiri) [reply app/views/layouts/_navbar.html.erb]: This is how to do the tab thing properly, according to the [docs](https://getbootstrap.com/docs/3.3/javascript/#tabs).

```js
$('#myTabs a[href="#profile"]').tab('show')
```

Comment (philippbayer) [reply app/views/layouts/_navbar.html.erb]: I've put in a new commit which gets rid of the method: get, and now it seems to work, even with the 'old' JS?

**Comment by philippbayer** [comment]
:man_shrugging: computers!

thank you for your help :)

**Pull request merged**